src/components/AR/AR-session.ts

```diff
@@ -0,0 +1,95 @@
+import { WonderlandEngine } from '@wonderlandengine/api';
+import { ARProvider } from './AR-provider.js';
+
+/**
+ * ARSession - master control for the AR session.
+ * - loads dependencies (aka providers)
+ * - handles global callbacks when AR session is started, ended
+ * - can end any running AR session.
+ * - renders AR button when the scene is loaded
+ */
+abstract class ARSession {
+    // tracking provider is basically a lib which has some tracking capabilities, so device native webXR, 8th Wall, mind-ar-js, etc
+    private static _trackingProviders: Array<ARProvider> = [];
+
+    // current running provider
+    private static _currentTrackingProvider: ARProvider | null = null;
+
+    public static readonly onARSessionReady: Array<() => void> = [];
+
+    public static readonly onSessionStarted: Array<(trackingProvider: ARProvider) => void> =
+        [];
+    public static readonly onSessionEnded: Array<(trackingProvider: ARProvider) => void> =
+        [];
+
+    private static _sceneHasLoaded = false;
+    private static _arSessionIsReady = false;
+
+    public static get arSessionReady() {
+        return this._arSessionIsReady;
+    }
+
+    /**
+     * Registers tracking provider. Makes sure it is loaded
+     * and hooks into providers onSessionStarted, onSessionLoaded events.
```

Comment:
Could you please add some typedoc documentation here for the parameters?
Doesn't need to too complex, just one liner for each is nice if we start uploading the typedoc on the website